### PR TITLE
Fix: Gradient Accumulation issue

### PR DESCRIPTION
**Pull request by NanoCode012**
Tests the upstream changes in `transformers` for GA fix and compare results. 

# Description

TODO:

- [ ] Check if Mamba Trainer needs to update loss
- [ ] Check if Orpo Trainer needs to update loss

Wait for:

- [x] New transformers release

## How has this been tested?

Testing and comparing loss in wandb.

### CODE REVIEW

**Comment by winglian** [comment]
also ran a manual check against ~https://github.com/axolotl-ai-cloud/axolotl/actions/runs/11418424220~
 https://github.com/axolotl-ai-cloud/axolotl/actions/runs/11443067437

**Comment by winglian** [comment]
@NanoCode012 I think we need to do some test runs and also verify the unsloth cross entropy loss since we had to swap that

**Comment by winglian** [comment]
I upgraded trl to the latest versions too for this, but seems like there are some changes that are breaking DPO

File: src/axolotl/core/trainer_builder.py [hunk 1]
```diff
@@ -1910,7 +1924,7 @@ def build(self, total_num_steps):
             dpo_trainer_kwargs["max_length"] = self.cfg.sequence_len
             dpo_trainer_kwargs["max_target_length"] = None
             dpo_trainer_kwargs["max_prompt_length"] = self.cfg.sequence_len
-            dpo_trainer_kwargs["generate_during_eval"] = True
+            dpo_trainer_kwargs["generate_during_eval"] = self.cfg.use_wandb
```
Comment (NanoCode012) [inline src/axolotl/core/trainer_builder.py:1958]: Is there a reason we specify to wandb only and not the others as well?

**Comment by winglian** [comment]
@NanoCode012 something isn't quite right with this PR, the loss for 8b with sample packing starts @ ~10, even without packing, the loss is ~10 for 8B

**Comment by NanoCode012** [comment]
I'll do some tests and report back. I'll move this to Draft in meantime.

**Comment by winglian** [comment]
@NanoCode012 upgrading to the transformers release that was pushed out today fixes the issue. I added a test too for a tiny model that also validates the loss

**Comment by winglian** [comment]
waiting on mulitgpu tests too here https://github.com/axolotl-ai-cloud/axolotl/actions/runs/11520374143
